src/Microsoft.AspNetCore.Hosting/Internal/WebHostOptions.cs

```diff
@@ -22,6 +22,7 @@ public WebHostOptions(IConfiguration configuration)
             }
 
             Application = configuration[WebHostDefaults.ApplicationKey];
+            EnvironmentApplication = configuration[WebHostDefaults.EnvironmentApplicationKey];
```

Comment:
The 2 concepts are:

- Startup assembly name - Where do I look for the startup class
- Application entry point -  What is the name of my application entry point

Are those 2 things always the same? I don't think they are.